engine/consensus/ingestion/engine.go

```diff
@@ -281,3 +254,93 @@ func (e *Engine) validateExpiry(guarantee *flow.CollectionGuarantee) error {
 
 	return nil
 }
+
+// validateGuarantors validates that the guarantors of a collection are valid,
+// in that they are all from the same cluster and that cluster is allowed to
+// produce the given collection w.r.t. the guarantee's reference block.
+func (e *Engine) validateGuarantors(guarantee *flow.CollectionGuarantee) error {
+
+	guarantors := guarantee.SignerIDs
+
+	if len(guarantors) == 0 {
+		return engine.NewInvalidInputError("invalid collection guarantee with no guarantors")
+	}
+
+	// get the clusters to assign the guarantee and check if the guarantor is part of it
+	snapshot := e.state.AtBlockID(guarantee.ReferenceBlockID)
+	clusters, err := snapshot.Epochs().Current().Clustering()
+	if err != nil {
+		return fmt.Errorf("could not get clusters: %w", err)
+	}
```

Comment:
If `guarantee.ReferenceBlockID` is an unknown block, this will makes it look like an unexpected (potentially fatal error).